### DISCUSSION TITLE
Support for Nintendo Switch Online N64 Controller.

### DISCUSSION THREAD

**Discussion by theksepyro** [simple64/simple64#452] [Ideas]
Seeing as the NSO N64 controller is a first party controller with drivers on windows and linux, it would be nice to have a configuration that works for it out of the box.

I attempted to map inputs in linux, but it regarded the z and down-c buttons as axes rather than buttons, and so they behaved strangely in games.

![image](https://github.com/user-attachments/assets/28fff56e-34a3-45b7-8922-ac94152eb4d3)

**Comment by loganmc10** [comment]
When you say behave strangely, what do you mean?

Simple64 uses SDL2 for input, so controller support actually depends on that project, there is no code in simple64 itself relating to specific controllers

**Comment by theksepyro** [comment]
I didn't document/characterize it's behavior very rigorously, so I don't know exactly how it was behaving. I tried playing super smash brothers and banjo-tooie mini games, and sometimes it seemed as though the down-c button was being held down, and sometimes it did not. I  checked to see if it was a hardware issue and it doesn't appear to be, and that led me to checking the mapping configuration, where you can see that down-c gets mapped differently than the rest of the c buttons.

In any case it sounds like I should raise this question with the SDL2 folks. Thanks for the consideration!

and to edit:
I have just compared this behavior to other projects that use the same input plugin and they seem to have a similar outcome where Z and down-C are treated as axes and get held down when pressed. This confirms that it's unrelated to this project specifically.

**Reply by theksepyro** [reply]
Just to follow up/close the loop, the SDL people acknowledged that the button mapping is weird,  but that it's working as intended. They suggested making changes in client software to address it.

https://github.com/libsdl-org/SDL/issues/10658

**Comment by loganmc10** [comment]
I don't have one of these controllers so I need really specific information if I am to fix it.

I understand that both the Z and C-down buttons are seen as axis. But are both having issues? Or just C down? Does the Z button currently behave correctly?

**Reply by dskrapates** [reply]
For me, the Z button seems to work as intended even though it's recognized as an axis.

It seems the C down button acts like it is constantly being pressed, but will also still function when it is pressed.

So in the game I tested for reference, C down has two functions. It will make the character crouch/prone when pressed, and when aiming with a sniper, it is supposed to zoom out the scope.

What is happening currently is, when C down is pressed the character will crouch/prone as intended. But, when aiming with the sniper, the scope will constantly zoom out even when the button isn't being pressed.

**Reply by theksepyro** [reply]
At the risk of adding clutter, I want to express that I have the same basic experience. Even though Z is mapped strangely it seems to be functioning appropriately in the games I've tested it on, but C down acts as thought it is always being held down.
